lib/Common/Io.hpp

```diff
@@ -265,12 +276,12 @@ class DInPin<pinNo, INPUT_PULLUP> : public IoPin
 
     /**
      * Read from digital input pin.
-     * 
+     *
      * @return Ditial input pin value.
      */
     int read() const
     {
-        return digitalRead(pinNo);
+        return (NC != pinNo) ? digitalRead(pinNo) : -1;

```

Comment:
Please follow Arduino: https://www.arduino.cc/reference/en/language/functions/digital-io/digitalread/
Recommended instead of -1 is LOW.

Reply:
fixed as suggested.